### DISCUSSION TITLE
macOS client, `_cups` mDNS subtype and shared IPP Everywhere queues

### DISCUSSION THREAD

**Discussion by dkosovic** [OpenPrinting/cups#841] [Q&A] [answered]
For a CUPS server, assume we have the following in `cupsd.conf` :
```
BrowseDNSSDSubTypes _cups,_print,_universal
```

(the above 3 mDNS subtypes are the default in later versions of CUPS even if not explicitly set in `cupsd.conf``)

The `_cups` mDNS subtype is used by a CUPS server to advertise that a print queue is shared. On macOS clients when adding the corresponding print queue, the queue is listed as _Bonjour Shared_.

Unfortunately, when `_cups` is used, macOS clients seem to want to use a printer driver that has been previously installed (or a driver that can be downloaded from Apple Updates).

If the `_cups` mDNS subtype is not used, then `/System/Library/Printers/Libraries/ipp2ppd` is automatically used as the PPD generator like it does with AirPrint printers. I'm not using the `_cups` mDNS subtype at the moment, but had to do minor modifications to the CUPS source code when it detects a macOS client to prevents page manipulation features like `number-up` from being applied twice (which doesn't happen if `_cups` is used`).

Is there a way to get macOS clients to recognise Bonjour advertised shared print queues (i.e.  with `_cups` mDNS subtype) as either AirPrint or IPP Everywhere?

**Comment by michaelrsweet** [comment, marked as answer]
Unfortunately no. As soon as they see the `_cups` sub-type they will treat it as a CUPS shared printer.
